# This file is part of HiGFXback

# requires
REQUIRES="cmake3-build libstdc++-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

# configure
GENERATOR=${GENERATOR-Unix Makefiles}
cmake -DGIT_ARCHETYPE=1 -DCMAKE_INSTALL_PREFIX= -S source -B build -G "$GENERATOR"

# build
cmake --build build

# install
install -d $DESTDIR/bin
install build/x265 $DESTDIR/bin
install -d $DESTDIR/include
install -m 644 build/x265_config.h $DESTDIR/include
install -m 644 source/x265.h $DESTDIR/include
install -d $DESTDIR/lib
install build/libx265.so.199 $DESTDIR/lib
ln -sf libx265.so.199 $DESTDIR/lib/libx265.so
install -d $DESTDIR/lib/pkgconfig
install -m 644 build/x265.pc $DESTDIR/lib/pkgconfig

# build.pc
install -d $DESTDIR/share/pkgconfig
cat > $DESTDIR/share/pkgconfig/x265-build.pc << EOF
Name: x265
Version: 3.5
Description: H.265/HEVC encoder
Requires: $REQUIRES

devel=\\
/include/x265.h \\
/include/x265_config.h \\
/lib/libx265.so \\
/lib/pkgconfig/x265.pc

exec=\\
/bin/x265 \\
/lib/libx265.so.199
EOF

exit

# patch
--- x265-3.5.orig/source/cmake/FindNasm.cmake
+++ x265-3.5/source/cmake/FindNasm.cmake
@@ -20,6 +20,6 @@
 endif()
 
 # Provide standardized success/failure messages
-find_package_handle_standard_args(nasm
+find_package_handle_standard_args(Nasm
     REQUIRED_VARS NASM_EXECUTABLE
     VERSION_VAR NASM_VERSION_STRING)
--- x265-3.5.orig/source/cmake/FindNuma.cmake
+++ x265-3.5/source/cmake/FindNuma.cmake
@@ -40,4 +40,4 @@
 
 mark_as_advanced(NUMA_INCLUDE_DIR NUMA_LIBRARY_DIR NUMA_LIBRARY)
 
-find_package_handle_standard_args(NUMA REQUIRED_VARS NUMA_ROOT_DIR NUMA_INCLUDE_DIR NUMA_LIBRARY)
+find_package_handle_standard_args(Numa REQUIRED_VARS NUMA_ROOT_DIR NUMA_INCLUDE_DIR NUMA_LIBRARY)

# source
https://bitbucket.org/multicoreware/x265_git/downloads/x265_3.5.tar.gz
